platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java

```diff
@@ -148,17 +163,19 @@ public AddressBook getInitialAddressBook() {
      * address book is the one to use.  All three address books, the configuration address book, the save state address
      * book, and the new address book to use are recorded in the address book directory.
      *
-     * @return the address book to use in the platform.
+     * @return the address books to use in the platform.
      */
     @NonNull
-    private AddressBook initialize() {
+    private InitializedAddressBooks initialize() {
         AddressBook candidateAddressBook;
+        final AddressBook previousAddressBook;
         if (useConfigAddressBook) {
             // settings.txt override to use config.txt address book
             logger.info(
                     STARTUP.getMarker(),
                     "Overriding the address book in the state with the address book from config.txt");
             candidateAddressBook = configAddressBook;
+            previousAddressBook = stateAddressBook;
```

Comment:
The previousAddressBook is set to null when there is a genesis state.  Does this need to check for genesis state and set to null?

Reply:
stateAddressBook will be null if this is genesis, right?

Reply:
https://github.com/hashgraph/hedera-services/blob/b1469cd45f595515cee9507e1d037a9659b8362d/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java#L826-L827

It looks like the config address book is used to initialize the genesis state.  

Since the `previousAddressBook` is a new concept, you get to determine its value in the event of genesis state.  :-)

Reply:
I simplified the logic a lot to:
`final AddressBook previousAddressBook = initialState.isGenesisState() ? null : stateAddressBook;`